### ISSUE TITLE
code block does not preserve HTML code when pasting

### ISSUE THREAD

**Issue by pwkip** [WordPress/gutenberg#27386]
**Describe the bug**
When pasting HTML code in the code block, the actual HTML gets rendered..

**To reproduce**
1. Create a code block
2. Paste in some HTML code
3. The rendered html is displayed instead of the source code.

**Expected behavior**
Preserve the code

**Screenshots**
<img src="https://user-images.githubusercontent.com/1150196/100709337-32878380-33ae-11eb-8948-c29c980bd3c7.gif" width="400">

**Happens on all tested editor versions:**
- WP 5.5.4 with Gutenberg 9.4.1
- WP 5.6-RC1 with default editor

**Happens on all tested OS and browsers:**
- Chrome on Ubuntu 20.04 Version 87.0.4280.66 (Official Build) (64-bit)
- Chrome on Windows 10
- Firefox on Windows 10

**Comment by Lewiscowles1986** [comment]
<details>
<summary>New HTML (output)</summary>

```
<code role="textbox" aria-multiline="true" aria-label="Code" class="block-editor-rich-text__editable rich-text" style="white-space: pre-wrap;" contenteditable="true"><strong data-rich-text-format-boundary="true">What whaaaat</strong></code>
```

</details>

This feels like it could be a browser bug with contenteditable, but if it is then it's one I think has existed for a while.

Should this just be a textarea?

Idk, it's working as expected on a few sites I've checked, but not in storybook for latest master.

<details>
<summary>Old HTML (output)</summary>

```
<code role="textbox" aria-multiline="true" aria-label="Code" class="rich-text block-editor-rich-text__editable" contenteditable="true" style="white-space: pre-wrap;">&lt;strong&gt;What whaaaat&lt;/strong&gt;</code>
```

</details>

**Comment by Lewiscowles1986** [comment]
I just removed the repo and re-pulled. Looks like this is fixed in master anyway. Check it out.

**Comment by pwkip** [comment]
I don't have time to test right away. I only know this is the behavior I get both on windows and linux (two completely different PC's and different browsers) with a fresh installation of WP 5.6-RC1. It happens for all blocks by the way. When I paste code from a text editor (VS code, notepad, gedit), Gutenberg automatically renders the HTML. This is actually pretty nice for paragraph blocks and so on, but doesn't make much sense for the code block.

Will try to test by pulling master later.

**Comment by tellthemachines** [comment]
Closing as this was fixed in #27236. Testing with latest master, I can no longer reproduce the issue. It is also fixed in 5.6 RC2 which is now available for testing.

**Comment by paaljoachim** [comment]
@tellthemachines Please see my comment here:
Copied formatted editor content from 2 paragraph blocks and a list block.
Opened Gmail and pasted with CTRL + SHIFT + V
Result is seeing the Gutenberg code.

https://github.com/WordPress/gutenberg/issues/11742#issuecomment-782776542
